testcontainers/src/core/client.rs

```diff
@@ -81,6 +90,12 @@ pub enum ClientError {
     InitExec(BollardError),
     #[error("failed to inspect exec command: {0}")]
     InspectExec(BollardError),
+    #[error("failed to copy to container: {0}")]
+    CopyToContainer(BollardError),
+    #[error("failed to tar data for container: {0}")]
+    TarError(io::Error),
+    #[error("failed to get the path name: {0}")]
+    PathNameError(String),
```

Comment:
I'd suggest to have one `CopyToContainerError` to encapsulate all the related errors into single one.

And here we can use transparent error

Reply:
That is a very good approach. I now put all errors related to prepare the data for container upload into a separate Error. Only the one `BollardError` which comes from the `docker.upload...` was handled separately, as it seems that all other bollard errors are also there.